lib/command/terraform/generate.rb

```diff
@@ -62,19 +62,39 @@ def generate_providers(terraform_app_dir)
         File.write(terraform_app_dir.join("providers.tf"), cpln_provider.to_tf)
       end
 
-      def recreate_terraform_app_dir
+      def required_cpln_provider
+        TerraformConfig::RequiredProvider.new(
+          name: "cpln",
+          org: config.org,
+          source: "controlplane-com/cpln",
+          version: "~> 1.0"
+        )
+      end
+
+      def cleaned_terraform_app_dir
         full_path = terraform_dir.join(config.app)
 
         unless File.expand_path(full_path).include?(Cpflow.root_path.to_s)
           Shell.abort("Directory to save terraform configuration files cannot be outside of current directory")
         end
 
-        FileUtils.rm_rf(full_path)
```

Comment:
We need to preserve `.terraform.lock.hcl` lock file so instead of recreating terraform application directory we remove all entries except lock file from each application folder